catroid/src/main/java/org/catrobat/catroid/content/actions/GoToRandomPositionAction.kt

```diff
@@ -20,40 +20,22 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-
-package org.catrobat.catroid.content.actions;
-
-import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction;
-
-import org.catrobat.catroid.common.ScreenValues;
-import org.catrobat.catroid.content.Sprite;
-
-public class GoToRandomPositionAction extends TemporalAction {
-
-	private Sprite sprite;
-	private float randomXPosition;
-	private float randomYPosition;
-
-	@Override
-	protected void update(float percent) {
-		randomXPosition = (float) Math.random()
-				* (ScreenValues.SCREEN_WIDTH + 1) - (ScreenValues.SCREEN_WIDTH / 2);
-		randomYPosition = (float) Math.random()
-				* (ScreenValues.SCREEN_HEIGHT + 1) - (ScreenValues.SCREEN_HEIGHT / 2);
-
-		sprite.look.setXInUserInterfaceDimensionUnit(randomXPosition);
-		sprite.look.setYInUserInterfaceDimensionUnit(randomYPosition);
-	}
-
-	public void setSprite(Sprite sprite) {
-		this.sprite = sprite;
-	}
-
-	public float getRandomXPosition() {
-		return this.randomXPosition;
-	}
-
-	public float getRandomYPosition() {
-		return this.randomYPosition;
-	}
+package org.catrobat.catroid.content.actions
+
+import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction
+import org.catrobat.catroid.common.ScreenValues.SCREEN_HEIGHT
+import org.catrobat.catroid.common.ScreenValues.SCREEN_WIDTH
+import org.catrobat.catroid.content.Sprite
+
+class GoToRandomPositionAction : TemporalAction() {
+    var sprite: Sprite? = null
+    var randomXPosition: Float = Math.random().toFloat() * SCREEN_WIDTH + 1 - SCREEN_WIDTH / 2
+    var randomYPosition: Float = Math.random().toFloat() * SCREEN_HEIGHT + 1 - SCREEN_HEIGHT / 2
+
+    override fun update(percent: Float) {
+        sprite?.apply {
+            look.xInUserInterfaceDimensionUnit = randomXPosition
```

Comment:
I could be wrong, but I think this can't work that way. 
randomXPosition and randomYPosition won't be updated.
Have you tried using the respective brick within a loop?

Reply:
If you inject the "Math.random" method, you could test corner cases ;)
Also, it does not do the same as the prev. method, you forgot one parenthesis :) But I think it doesn't make a big difference

Reply:
You are absolutely right in your assumption. randomXPosition and randomYPosition won't be updated. I will fix it and inform you afterwards.